### PR TITLE
fix: mis-match schema config

### PR DESCRIPTION
**Pull request by almog8k**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✔                                                                      |
| New feature     |✖                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✖                                                                       |

### CODE REVIEW

**Comment by CptSchnitz** [comment]
waiting for merge of #47

**Comment by almog8k** [comment]
> waiting for merge of #47

merged

File: src/pages/createConfig/step1GeneralInfo/step1GeneralInfo.tsx [hunk 1]
```diff
@@ -36,6 +33,21 @@ export const Step1GeneralInfo: React.FC = () => {
     dispatch({ type: 'SET_VALIDATION_RESULT', step: 'step1', payload: isValid });
   }, [isValid, configName, schemaId, description, dispatch]);
 
+  useEffect(() => {
+    if (latestConfig) {
+      if (latestConfig.schemaId !== schemaId) {
+        setError(
+          'root',
+          { message: `mis-match schema to config name (the matching schema is: "${latestConfig.schemaId}")`, type: 'onChange' },
```
Comment (CptSchnitz) [inline src/pages/createConfig/step1GeneralInfo/step1GeneralInfo.tsx:41]:
```suggestion
          { message: `mismatch schema to config name (the matching schema is: "${latestConfig.schemaId}")`, type: 'onChange' },
```

File: src/pages/createConfig/step1GeneralInfo/step1GeneralInfo.tsx [hunk 2]
```diff
@@ -36,6 +33,21 @@ export const Step1GeneralInfo: React.FC = () => {
     dispatch({ type: 'SET_VALIDATION_RESULT', step: 'step1', payload: isValid });
   }, [isValid, configName, schemaId, description, dispatch]);
 
+  useEffect(() => {
+    if (latestConfig) {
+      if (latestConfig.schemaId !== schemaId) {
```
Comment (CptSchnitz) [inline src/pages/createConfig/step1GeneralInfo/step1GeneralInfo.tsx:38]: reverse the logic to avoid nesting

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

**Review by CptSchnitz — APPROVED** [review]

**Pull request merged**